lib/ansible/plugins/lookup/aws_ssm.py

```diff
@@ -45,130 +48,150 @@
 
 - name: lookup ssm parameter store with all options.
-  debug: msg="{{ lookup('aws_ssm', 'Hello', 'decrypt=false', 'region=us-east-2', 'aws_profile=myprofile') }}"
+  debug: msg="{{ lookup('aws_ssm', 'Hello', decrypt=false, region=us-east-2, aws_profile=myprofile') }}"
 
 - name: return a dictionary of ssm parameters from a hierarchy path
-  debug: msg="{{ lookup('aws_ssm', '/PATH/to/params', 'region=ap-southeast-2', 'bypath', 'recursive=true' ) }}"
+  debug: msg="{{ lookup('aws_ssm', '/PATH/to/params', region=ap-southeast-2, bypath=true, recursive=true' ) }}"
 
 - name: return a dictionary of ssm parameters from a hierarchy path with shortened names (param instead of /PATH/to/param)
-  debug: msg="{{ lookup('aws_ssm', '/PATH/to/params', 'region=ap-southeast-2', 'shortnames', 'bypath', 'recursive=true' ) }}"
+  debug: msg="{{ lookup('aws_ssm', '/PATH/to/params', region=ap-southeast-2, shortnames, bypath=true, recursive=true ) }}"
 
 - name: Iterate over a parameter hierarchy
   debug: msg='key contains {{item.Name }} with value {{item.Value}} '
   with_aws_ssm:
-    - '/TEST/test-list'
-    - 'region=ap-southeast-2'
-    - 'bypath'
+    - '/TEST/test-list region=ap-southeast-2, bypath'
 '''
+# FIXME the last one is probably not true yet. 
 
-from ansible.module_utils.ec2 import HAS_BOTO3
+from ansible.utils.display import Display
+from ansible.module_utils.ec2 import HAS_BOTO3, boto3_tag_list_to_ansible_dict
 from ansible.errors import AnsibleError
 from ansible.plugins.lookup import LookupBase
 from ansible.module_utils.parsing.convert_bool import boolean
+import pdb
+
+from ansible.utils.display import Display
+
+try:
+    from __main__ import display
+except ImportError:
+    from ansible.utils.display import Display
+    display = Display()
 
 try:
     from botocore.exceptions import ClientError
+    import botocore
     import boto3
 except ImportError:
     pass  # will be captured by imported HAS_BOTO3
 
 
+def _boto3_conn(region, credentials):
+    if 'boto_profile' in credentials:
+        boto_profile = credentials.pop('boto_profile')
+    else:
+        boto_profile = None
+
+    try:
+        connection = boto3.session.Session(profile_name=boto_profile).client('ssm', region, **credentials)
+    except (botocore.exceptions.ProfileNotFound, botocore.exceptions.PartialCredentialsError) as e:
+        if boto_profile:
+            try:
+                connection = boto3.session.Session(profile_name=boto_profile).client('ssm', region)
+            except (botocore.exceptions.ProfileNotFound, botocore.exceptions.PartialCredentialsError) as e:
+                raise AnsibleError("Insufficient credentials found.")
+        else:
+            raise AnsibleError("Insufficient credentials found.")
+    return connection
+
+
 class LookupModule(LookupBase):
-    def run(self, terms, variables, **kwargs):
+    def run(self, terms, variables, boto_profile=None, aws_secret_key=None, aws_access_key=None,
```

Comment:
traditionally, variables is a keyword arg.  I don't think it matters in our paticular code base but we should keep it consistent with other lookup plugins.

Reply:
hmmm....
```bash
[mikedlr@localhost lookup]$ egrep -nH -e 'def run.*variables' * | wc
     47     281    3087
```
vs
```bash
[mikedlr@localhost lookup]$ egrep -nH -e 'def run.*variables=' * | wc
     28     167    1874
```

and all of the other aws* lookups don't do it as keyword.  I think we'd benefit from doing a PR to change almost all of these since I agree about keeping consistent; it's one of the most difficult things about the Ansible codebase.  For now I'm going to do it for this module only.